lms/djangoapps/discussion/views.py

```diff
@@ -214,31 +221,25 @@ def forum_form_discussion(request, course_key):
     nr_transaction = newrelic.agent.current_transaction()
 
     course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=True)
-    course_settings = make_course_settings(course, request.user)
-
-    user = cc.User.from_django_user(request.user)
-    user_info = user.to_dict()
+    if request.is_ajax():
```

Comment:
The high level description is that I moved all of the HTML rendering logic into the new fragment view, both from this view (`forum_form_discussion`) and from the single thread view (`single_thread`). I couldn't remove these two views entirely though because they also function as the REST APIs when receiving requests for JSON, so I refactored the logic out into shared helper functions where they were needed by both. This refactoring explains the reduction in number of queries in the test you commented on.